src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): {} {
+  const groups: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const groupKey = String(item[key]);
+
+    if (groupKey === undefined) {
+      return;
+    }
+
+    if (!groups[groupKey]) {
+      groups[groupKey] = items.filter((el) => el[key] === item[key]);
+    }
```

Comment:
Now you iterate through item on every iteration. It is bad solution
you can:
* create an empy array as value of key
* push item into this array after check